pbt-libraries/americium/src/test/java/challenges/binheap/Heap.java

```diff
@@ -0,0 +1,43 @@
+package challenges.binheap;
+
+// NOTE: adapted from the Jqwik challenge version, an invariant check in
+// the spirit of the QuickCheck / Smallcheck version has been added in too.
+
+import cyclops.data.tuple.Tuple3;
+
+class Heap {
+
+    final int head;
+    final Heap left;
+    final Heap right;
+
+    Heap(int head, Heap left, Heap right) {
+        this.head = head;
+        this.left = left;
+        this.right = right;
+
+        checkInvariant();
+    }
+
+    void checkInvariant() {
+        if (null != left) {
+            assert head <= left.head;
+
+            left.checkInvariant();
```

Comment:
I should remove this, as `Heap` is an immutable type - so what's the point of recursively checking the invariant, it is already established by construction for the left and right children.

Reply:
Done.